File: Server_Nodejs/models/sanpham.js

// Trong file sanpham.js
import { Model, DataTypes } from 'sequelize';
import mydb from '../config/connect_mysql.js';
import DanhMuc from './danhmuc.js';
import NguoiDung from './nguoidung.js';
import Image from './image.js';

const SanPham = mydb.define('sanpham', {
    id_sanpham: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    ten_sanpham: {
        type: DataTypes.STRING
    },
    id_nguoidung: {
        type: DataTypes.INTEGER
    },
    id_danhmuc: {
        type: DataTypes.INTEGER
    },
    ngay_tao: {
        type: DataTypes.INTEGER
    },

    gia: {
        type: DataTypes.DOUBLE
    },
    mo_ta: {
        type: DataTypes.STRING
    },
    sdt: {
        type: DataTypes.STRING
    },
    diachi: {
        type: DataTypes.STRING
    },
    status: {
        type: DataTypes.BOOLEAN
    },
    // createdAt: {
    //     field: 'created_at',
    //     type: DataTypes.DATE
    // },
    // updatedAt: {
    //     field: 'updated_at',
    //     type: DataTypes.DATE
    // }
}, {
    tableName: 'sanpham',
    timestamps: false
});

// Thiết lập quan hệ giữa các bảng
SanPham.belongsTo(DanhMuc, { foreignKey: 'id_danhmuc' });
SanPham.belongsTo(NguoiDung, { foreignKey: 'id_nguoidung' });
SanPham.belongsTo(Image, { foreignKey: 'id_sanpham' });

export default SanPham;

File: second_hand_store/lib/models/sanpham.dart
class SanPham {
  int? idSanpham;
  String? tenSanpham;
  int? idNguoidung;
  int? idDanhmuc;
  int? ngayTao;
  double? gia;
  String? moTa;
  String? sdt;
  String? diachi;
  bool? status;
  String? danhmuc;
  String? nguoidung;
  List<String>? imageArr;

  SanPham(
      {this.idSanpham,
      this.tenSanpham,
      this.idNguoidung,
      this.idDanhmuc,
      this.ngayTao,
      this.gia,
      this.moTa,
      this.sdt,
      this.diachi,
      this.status,
      this.danhmuc,
      this.nguoidung,
      this.imageArr});

  SanPham.fromJson(Map<String, dynamic> json) {
    idSanpham = json['id_sanpham'];
    tenSanpham = json['ten_sanpham'];
    idNguoidung = json['id_nguoidung'];
    idDanhmuc = json['id_danhmuc'];
    ngayTao = json['ngay_tao'];
    gia = json['gia'];
    moTa = json['mo_ta'];
    sdt = json['sdt'];
    diachi = json['diachi'];
    status = json['status'];
    danhmuc = json['danhmuc'];
    nguoidung = json['nguoidung'];
    imageArr = json['imageArr'].cast<String>();
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id_sanpham'] = idSanpham;
    data['ten_sanpham'] = tenSanpham;
    data['id_nguoidung'] = idNguoidung;
    data['id_danhmuc'] = idDanhmuc;
    data['ngay_tao'] = ngayTao;
    data['gia'] = gia;
    data['mo_ta'] = moTa;
    data['sdt'] = sdt;
    data['diachi'] = diachi;
    data['status'] = status;
    data['danhmuc'] = danhmuc;
    data['nguoidung'] = nguoidung;
    data['imageArr'] = imageArr;
    return data;
  }
}
